### PR TITLE
Fix (and add tests for) BaseElement.__ne__

### LINKED ISSUE #490: 🐞  Double negative on not equal function on BaseElement
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

https://github.com/microsoft/electionguard-python/blob/6bf466f3cf8b3f47cd29e1c549f86ad1758b6221/src/electionguard/group.py#L32-L38

### Expected Behavior

The not equal (`__ne__`) function should correctly handle not equal. 

### Steps To Reproduce

_No response_

### Environment

```markdown
- OS: Any
```

### Anything else?

Please add a test in `test_group.py` to ensure this bug does not reappear.

### PR DESCRIPTION
**Pull request by ronnie-llamado**
[//]: # (🚨 Please review the CONTRIBUTING.md in this repository. 💔Thank you!)

### Issue
Fixes #490 

### Description
Fix double negative inside of `BaseElement.__ne__`

### Testing
`make coverage`

### CODE REVIEW

File: src/electionguard/group.py
Comment (keithrfung) [inline src/electionguard/group.py:34]: How does this react to items that aren't `int` or `BaseElement`? 

For example:
string = str(q)
self.assertEqual(q, string)

Comment (ronnie-llamado) [reply src/electionguard/group.py:34]: Since strings will return `False` in `__eq__`, any comparison against strings will not be equal in this setup.

Here's the StackOverflow inspiration: https://stackoverflow.com/questions/4352244/should-ne-be-implemented-as-the-negation-of-eq

Should the example you provided be True, or was that just for example purposes? `ElementModQ(1) == "1"`?

Comment (keithrfung) [reply src/electionguard/group.py:34]: That was for example purposes to make sure we can remove the `isinstance` call.

**Review by keithrfung — APPROVED** [review]
Perfect example of me after my booster shot giving approval.

@SteveMaier-IRT Can I get a second approval?

**Review by SteveMaier-IRT — APPROVED** [review]

**Pull request merged**
